### PR TITLE
Allows Creatures To Change Eye Angles and Distances Through Evolution

### PR DESCRIPTION
**Pull request by UnknownEvil**
I've checked and confirmed that they work this time, trust me, 
or check yourself, either way works.

Edit: Wait for 1 last commit, I managed to make a mistake. Whoops.
Edit: Commit has been made.

### CODE REVIEW

**Comment by UnknownEvil** [comment]
I'll see if I can allow them to change their number of eyes as well.

**Comment by PRabahy** [comment]
Grr... I didn't realize I was commenting about a specific commit. My bad.

**Comment by PRabahy** [comment]
I just pulled the changes down and I'm testing them now.

**Comment by PRabahy** [comment]
If anyone else is wondering. Here is how it looks with various numbers of eyes. Its not perfect, but IMO, its good enough to merge.

image

image

image

**Comment by UnknownEvil** [comment]
Yeah, sorry about the GUI things and some of the hard-coded stuff, I didn't want to have to re-work the whole brain, and the whole GUI.

**Comment by PRabahy** [comment]
Hm... As I was running this, I noticed that board.colliders doesn't appear to be getting cleaned up properly. I don't see any code that should have messed with that, but after a couple hundred years I only had ~200 creatures alive and over 1000 entries in that. I'm testing with master right now and don't see the same behavior.

File: src/evolv/io/Creature.java [hunk 1]
```diff
@@ -340,6 +346,10 @@ public void reproduce(double babySize, double timeStep) {
 					newSaturation += parent.getSaturation() / parentsTotal;
 					newBrightness += parent.getBrightness() / parentsTotal;
 					newMouthHue += parent.mouthHue / parentsTotal;
+					for (int j = 0; j < NUM_EYES; j++) {
+						newEyeAngles[j] = parent.eyes.get(j).angle / parentsTotal;
+						newEyeDistances[j] = parent.eyes.get(j).distance / parentsTotal;
```
Comment (PRabahy) [inline src/evolv/io/Creature.java]: I think this math is wrong. If a creature, has 3 parents, its eye angle and distance is just going to be .333 of its last parent. I think just changing the = to += will fix it.

Comment (UnknownEvil) [inline src/evolv/io/Creature.java]: Oh yeah, whoops, Didn't catch that. Sorry.

Comment (UnknownEvil) [inline src/evolv/io/Creature.java]: As for the incorrect cleaning, I've no idea what's causing that.

**Comment by UnknownEvil** [comment]
That over sized thing doesn't make sense, population is meant to be exactly what that is.
thing

**Comment by UnknownEvil** [comment]
But, I can confirm what you're getting...

thing

**Comment by PRabahy** [comment]
I'm going to merge this because I want to do some work tonight that builds on top of this. Hopefully I'll figure out the bug in the process and squash it.

**Pull request merged**